src/domain/accounts/index.types.d.ts

```diff
@@ -113,15 +128,37 @@ type AccountValidator = {
   validateWalletForAccount(wallet: Wallet): true | ValidationError
 }
 
+type listAccountIdsByCustomFieldArgs = {
+  field: string
+  value: AccountCustomFieldValues
+}
+
+type PersistNewCustomFieldsArgs = {
+  accountId: AccountId
+  createdByUserId: UserId
+  customFields: { [k: string]: AccountCustomFieldValues }
+}
+
 interface IAccountsRepository {
   listUnlockedAccounts(): Promise<Account[] | RepositoryError>
   findById(accountId: AccountId): Promise<Account | RepositoryError>
   findByUserId(userId: UserId): Promise<Account | RepositoryError>
   findByUsername(username: Username): Promise<Account | RepositoryError>
   listBusinessesForMap(): Promise<BusinessMapMarker[] | RepositoryError>
+  listByIds(accountIds: AccountId[]): Promise<Account[] | RepositoryError>
   update(account: Account): Promise<Account | RepositoryError>
 }
 
+interface IAccountCustomFieldsRepository {
+  findById(accountId: AccountId): Promise<AccountCustomFields | RepositoryError>
+  listAccountIdsByCustomField(
```

Comment:
Should be `listByCustomField`

Reply:
could you please check the query? (it is an aggregate) really is only account ids query, there is no way to query the additional fields without do another query. do we want to do that?

Reply:
Updated with the new model https://github.com/GaloyMoney/galoy/pull/1465/commits/22c9f29082861a0a3ae06bfd1c0c82247e53368b